lua/telescope/actions/init.lua

```diff
@@ -225,6 +225,49 @@ actions.run_builtin = function(prompt_bufnr)
   require('telescope.builtin')[entry.text]()
 end
 
+-- Maybe upstream in plenary. TJ was already working on a new Path module
+-- Or maybe tami5 is interested.
+local create_file = function(file)
+  vim.loop.fs_open(file, "w", 420, vim.schedule_wrap(function(err, fd)
+    if err then
+      vim.api.nvim_err_writeln('Could not create file: ' .. file)
+    else
+      vim.loop.fs_close(fd)
+    end
+  end))
+end
+
+actions.create_file = function(prompt_bufnr)
+  local file = vim.fn.input('Create file or directory(add / at the end of file). Enter name: ')
+  if file == "" then return end
+
+  local fpath = actions.get_selected_entry(prompt_bufnr).path
+  fpath = string.sub(fpath, -1) == '/' and vim.fn.fnamemodify(fpath, ':h:h') or vim.fn.fnamemodify(fpath, ':h')
+  fpath = fpath .. '/' .. file
+
+  actions.close(prompt_bufnr)
+
+  if string.sub(fpath, -1) ~= '/' then
```

Comment:
path.sep

Also, I think `fs_stat` can tell if it's a dir?

Reply:
Yes but if i recall correctly that is about creating a file/dir and if the user left a `/` at the end we wanna create a dir. But we should definitly use `path.sep` thanks